code/app/src/main/java/dev/tonholo/study/pokedex/screens/pokemonList/PokemonListViewModel.kt

```diff
@@ -40,15 +47,90 @@ class PokemonListViewModel @Inject constructor(
     }
 
     fun loadPokemonListPaginated() {
+        Log.d(TAG, "loadPokemonListPaginated() called")
+        Log.d(TAG, "loadPokemonListPaginated: currentPage = $currentPage")
+        Log.d(TAG, "loadPokemonListPaginated: pokemonList.size = ${pokemonList.value.size}")
+        Log.d(TAG, "loadPokemonListPaginated: isEndReached = $isEndReached")
+        Log.d(TAG, "loadPokemonListPaginated: isSearching = $isSearching")
+
+        val offset = PAGE_SIZE
+        val cursor = currentPage * PAGE_SIZE
         viewModelScope.launch {
-            isLoading.value = true
-            val params = GetPokemonListUseCase.Params(
+            getPokemonListFromDatabaseUseCase(GetPokemonListFromDatabaseUseCase.Params(
+                cursor = cursor,
+                offset = offset,
+            ))
+                .map(::mapFromDatabase)
+                .map(::mapFromRemote)
+                .map(::cacheRemoteResponse)
+                .distinctUntilChanged()
+                .collectLatest { resource ->
+                    Log.d(TAG, "collectLatest() called with: resource = $resource")
+                    Log.d(TAG, "collectLatest: pokemonList.size = ${pokemonList.value.size}")
+                    Log.d(TAG, "collectLatest: currentPage = $currentPage")
+
+                    when (resource) {
+                        is Resource.Error -> {
+                            loadingError.value = "${resource.message}\n${resource.throwable?.localizedMessage}"
+                            isLoading.value = false
+                        }
+                        is Resource.Loading -> isLoading.value = true
+                        is Resource.Success -> {
+                            if (resource.data.isNotEmpty()) {
+                                pokemonList.value += resource.data
```

Comment:
Compared to the other solution there are an extra `if` and `map` that occurs. Is that really different between two solutions? I was with the impression that this point of execution (returning) would be the same
```kotlin
if (!pokemonList.value.any { it.number == results.first().number }
    && !pokemonList.value.any { it.number == results.last().number }) {
               pokemonList.value += results.map { pokemonListResult ->
               ....
```
https://github.com/rafaeltonholo/study-pokedex/pull/4/files#diff-a65220b6aecfbac6cdf7a931dd93fb98ad71998f672792569fb15ff4e9a0dc16R126